File: Difficulty: Easy/Remove Duplicates Sorted Array/remove-duplicates-sorted-array.cpp
class Solution {
  public:
    // Function to remove duplicates from the given array.
    vector<int> removeDuplicates(vector<int> &arr) {
        // code here
        
        int index = 1;
        
        for(int i = 1; i < arr.size(); i++) {
            if(arr[i] != arr[i - 1]) {
                arr[index] = arr[i];
                index++;
            } 
        }
        arr.resize(index);      // Removes elements from the end
        return arr;
    }
};

File: Difficulty: Easy/Remove Duplicates Sorted Array/remove-duplicates-sorted-array.java
class Solution {
    ArrayList<Integer> removeDuplicates(int[] arr) {
        ArrayList<Integer> ans = new ArrayList<>();
        Set<Integer> st = new LinkedHashSet<>();
        
        for(int x : arr) {
            st.add(x);
        }
        for(int v : st) {
            ans.add(v);
        }
        
        return ans;
        
    }
}
